From 32a350ae752ab40a085315f5cbd7d455f8ae12e1 Mon Sep 17 00:00:00 2001
From: TALU <taluemail272@gmail.com>
Date: Wed, 17 Nov 2021 19:58:38 +0000
Subject: [PATCH 1/2] libfimg5x: Fix a -Wunreachable-code-loop-increment
 compilation error

hardware/samsung_slsi/exynos/libfimg5x/FimgExynos5.cpp:60:58: error: loop will run at most once (loop increment never executed) [-Werror,-Wunreachable-code-loop-increment]
    for(int i = m_curFimgV4xIndex; i < NUMBER_FIMG_LIST; i++) {
                                                         ^~~

Note that I'm not entirely sure if this change is right.

Change-Id: I062fecdec6b0a00531311d9a026011a229cf755f
---
 libfimg5x/FimgExynos5.cpp | 24 +++++++++++-------------
 1 file changed, 11 insertions(+), 13 deletions(-)

diff --git a/libfimg5x/FimgExynos5.cpp b/libfimg5x/FimgExynos5.cpp
index 4dda956..799037c 100644
--- a/libfimg5x/FimgExynos5.cpp
+++ b/libfimg5x/FimgExynos5.cpp
@@ -57,30 +57,28 @@ FimgApi *FimgV4x::CreateInstance()
 {
     FimgApi *ptrFimg = NULL;
 
-    for(int i = m_curFimgV4xIndex; i < NUMBER_FIMG_LIST; i++) {
-        if (m_ptrFimgApiList[i] == NULL)
-            m_ptrFimgApiList[i] = new FimgV4x;
+    while (m_curFimgV4xIndex < NUMBER_FIMG_LIST) {
+        if (m_ptrFimgApiList[m_curFimgV4xIndex] == NULL)
+            m_ptrFimgApiList[m_curFimgV4xIndex] = new FimgV4x;
 
-        if (m_ptrFimgApiList[i]->FlagCreate() == false) {
-            if (m_ptrFimgApiList[i]->Create() == false) {
-                PRINT("%s::Create(%d) fail\n", __func__, i);
+        if (m_ptrFimgApiList[m_curFimgV4xIndex]->FlagCreate() == false) {
+            if (m_ptrFimgApiList[m_curFimgV4xIndex]->Create() == false) {
+                PRINT("%s::Create(%d) fail\n", __func__, m_curFimgV4xIndex);
                 goto CreateInstance_End;
-            }
-            else
+            } else
                 m_numOfInstance++;
         }
 
-        if (i < NUMBER_FIMG_LIST - 1)
-            m_curFimgV4xIndex = i + 1;
+        if (m_curFimgV4xIndex < NUMBER_FIMG_LIST - 1)
+            m_curFimgV4xIndex++;
         else
             m_curFimgV4xIndex = 0;
 
-        ptrFimg = m_ptrFimgApiList[i];
+        ptrFimg = m_ptrFimgApiList[m_curFimgV4xIndex];
         goto CreateInstance_End;
     }
 
-CreateInstance_End :
-
+CreateInstance_End:
     return ptrFimg;
 }
 
-- 
2.34.1

